### DISCUSSION TITLE
How are tiff tags resolutionx and y set when converting svs to tif?

### DISCUSSION THREAD

**Discussion by miriam-groeneveld** [libvips/libvips#2059] [Q&A]
I'm using pyvips/openslide to convert svs images to tif. The svs file has properties openslide.mppx = 0.2525 and openslide.mppy = 0.2525. The resulting tif file however has a resolution of 1 (tiff tags ResolutionX and ResolutionY). That doesn't seem correct? The used code:
`
image = converter.Image.new_from_file(
    str(path.absolute()), access="sequential"
)

image.write_to_file(
    str(new_file_name.absolute()),
    tile=True,
    pyramid=True,
    bigtiff=True,
    compression="jpeg",
    Q=70,
)`

pyvips 2.1.12
openslide-python 1.1.2

**Comment by jcupitt** [comment]
Hello @miriam-groeneveld,

This was improved in 8.10, so it should work. Are you using an old version of libvips?

I see:

```
$ vips --version
vips-8.10.2-Mon Oct 12 16:43:59 UTC 2020
$ vipsheader -f xres CMU-2.svs 
2004.0080160320642
```

**Comment by miriam-groeneveld** [comment]
Hi John,
I think I have the right version:
```
$ vips --version   
vips-8.7.4-Fri Jan 18 17:07:38 UTC 2019   
$ vipsheader -f xres  AperioJP2K-33003-1.svs   
1

```

In this case I'm using the Aperio testdata from http://openslide.cs.cmu.edu/download/openslide-testdata/Aperio/ but I run into the same issue with our own imagefiles (also from Aperio).

**Comment by jcupitt** [comment]
You need 8.10 and you have 8.7 -- openslide resolution handling was improved just a few months ago.

**Comment by miriam-groeneveld** [comment]
Ah, sorry about that, you are correct. I'll get libvips updated to 8.10 then, thanks for the help!

**Comment by miriam-groeneveld** [comment]
We are using the python 3.8 docker image, which hasn't accepted VIPS 8.10 into their stable version yet (https://tracker.debian.org/pkg/vips). I can solve the problem by adding a image.copy() with xres and yres calculated from the openslide properties as a work around until 8.10 has been added. Would that work?

**Comment by jcupitt** [comment]
Yes, that should be fine.

You can also set xres/yres in tiffsave, eg.:

```python
image.write_to_file("x.tif", xres=2000, yres=2000)
```
